### PR TITLE
change to ResetValidationTokenSource to guard against a disposed validationTokenSource

### LINKED ISSUE #2310: [BUG] Cancel being called on a disposed CancellationTokenSource
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

The app is a fairly basic app that renders questionnaires (from JSON) into popups, or pages, that can be filled out. This is used as a package in other apps we have. Inside of the code is a sample app. When you load up a form and click into a textbox, the app crashes, with the following being printed to the console. From the best I can tell, it looks like Cancel() is called on a disposed cancellationtokensource. The app uses validation behaviors based on the question type.

**System.ObjectDisposedException:** 'Loading...'

**System.ObjectDisposedException:** 'The CancellationTokenSource has been disposed.'

[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: The CancellationTokenSource has been disposed.
[mono-rt]    at System.Threading.CancellationTokenSource.ThrowIfDisposed()
[mono-rt]    at System.Threading.CancellationTokenSource.Cancel(Boolean throwOnFirstException)
[mono-rt]    at System.Threading.CancellationTokenSource.Cancel()
[mono-rt]    at CommunityToolkit.Maui.Behaviors.ValidationBehavior.ResetValidationTokenSource(CancellationTokenSource newTokenSource)
[mono-rt]    at CommunityToolkit.Maui.Behaviors.ValidationBehavior.UpdateStateAsync(VisualElement view, ValidationFlags flags, Boolean isForced, Nullable`1 parentToken)
[mono-rt]    at CommunityToolkit.Maui.Behaviors.ValidationBehavior.OnViewPropertyChanged(VisualElement sender, PropertyChangedEventArgs e)
[mono-rt]    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
[mono-rt]    at Android.App.SyncContext.<>c__DisplayClass2_0.<Post>b__0() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:line 36
[mono-rt]    at Java.Lang.Thread.RunnableImplementor.Run() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:line 37
[mono-rt]    at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Java.Lang.IRunnable.cs:line 84
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 22

### Expected Behavior

You should be able to open the form, click into a textbox, and fill out the form w/o issue.

### Steps To Reproduce

1. Open solution, navigate to main branch, and build and run for Android.
2. Once app loads, select button "Fill Out Form".
3. Click into the "Number of Milk Crates Delivered" entry and app will crash.

### Link to public reproduction project repository

https://github.com/matt-bartholomew/MobileForms

### Environment

- .NET MAUI CommunityToolkit: 9.0.3
- OS: Android XT3 running Android 9
- .NET MAUI: 9.0.0-rc.1.24453.9

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by matt-bartholomew**
…

 ### Description of Change ###

 
Reset validation token source may be called on a disposed validationTokenSource, resulting in an objectdisposedexception.

 ### Linked Issues ###
 

 - Fixes #2310 

 ### PR Checklist ###
 
 - [X] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [X] Has samples (if omitted, state reason in description)
 - [X] Rebased on top of `main` at time of PR
 - [X] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###
Unsure how to add tests - never done testing before.
I think "Has samples" is good/checked, as the issue has a sample it can be tested against.
Honestly I dont understand the comment behind documentation created or updated.

### CODE REVIEW

**Review by brminnick — APPROVED** [review]
Thanks @matt-bartholomew!

**Comment by matt-bartholomew** [comment]
@brminnick Going to be honest - my first time contributing to opensource stuff and to the communitytoolkit. What's the timeline between this change getting approved, merged, and then actually published in an updated package? I don't know if that information is public anywhere, I just can't find it.

**Comment by matt-bartholomew** [comment]
@brminnick Currently failing 3/6 statuses/checks. Not sure that these failures are anything within my control/due to my changes. Can you have a look and just let me know? Thanks!

**Comment by bijington** [comment]
Thank you for the submission.

The pipeline errors don't appear to be caused by your changes. Well sort those at some point. We've been trying to ship .NET 9.0 support so a release will likely fall on when that gets completed. 

We don't have strict timelines for this stuff. Just as and when there is enough to ship

File: src/CommunityToolkit.Maui/Behaviors/Validators/ValidationBehavior.shared.cs [hunk 1]
```diff
@@ -421,8 +421,15 @@ void UpdateStyle(in VisualElement view, bool isValid)
 
 	void ResetValidationTokenSource(CancellationTokenSource? newTokenSource)
 	{
-		validationTokenSource?.Cancel();
-		validationTokenSource = newTokenSource;
+		try
+		{
+			validationTokenSource?.Cancel();
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Behaviors/Validators/ValidationBehavior.shared.cs:426]: I believe we should dispose the tokenSource as well

```suggestion
			validationTokenSource?.Cancel();
			validationTokenSource?.Dispose();
```

**Review by pictos — CHANGES_REQUESTED** [review]

File: src/CommunityToolkit.Maui/Behaviors/Validators/ValidationBehavior.shared.cs (hunk 1 above)
Comment (matt-bartholomew) [reply src/CommunityToolkit.Maui/Behaviors/Validators/ValidationBehavior.shared.cs:426]: @pictos I made this change.

**Comment by pictos** [comment]
@matt-bartholomew can you take at your end and see if the project builds?

**Comment by matt-bartholomew** [comment]
> @matt-bartholomew can you take at your end and see if the project builds?

@pictos I built and tested on my end on windows using the sample app from the linked issue. I had no issue running or anything like that.

**Comment by pictos** [comment]
@matt-bartholomew are you on .net 8 or 9? Just to confirm if the issue is on our pipeline

**Comment by matt-bartholomew** [comment]
> @matt-bartholomew are you on .net 8 or 9? Just to confirm if the issue is on our pipeline

@pictos 8.

**Comment by matt-bartholomew** [comment]
@pictos @brminnick @bijington I don't think some of these pipeline failures are due to anything I introduced. Can one of you look and let me know? When would I be able to see this in a release?

**Comment by pictos** [comment]
@matt-bartholomew no worries about it. PRs will **not** be merged until we figure out .NET 9 support

**Comment by bijington** [comment]
And we should be very close now!

**Review by brminnick — APPROVED** [review]
Thanks @matt-bartholomew!

**Pull request merged**